source/portable/mbedtls/core_pkcs11_mbedtls.c

```diff
@@ -3209,6 +3209,7 @@ CK_DECLARE_FUNCTION( CK_RV, C_GetAttributeValue )( CK_SESSION_HANDLE hSession,
                             LogError( ( "Failed to parse attribute template. "
                                         "Received a buffer smaller than CK_OBJECT_CLASS." ) );
                             xResult = CKR_BUFFER_TOO_SMALL;
+                            pTemplate[ iAttrib ].ulValueLen = CK_UNAVAILABLE_INFORMATION;
```

Comment:
One minor suggestion, how do you think if we move these all to line 3192 as initialization?

Reply:
Then we can avoid lots of manual setting in error case.

Reply:
`ulValueLen` is also used to pass vValue length with some attribute type. Therefore, it should be set to CK_UNAVAILABLE_INFORMATION only with some cases.